dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/AITests/OpenAIModelTests.cs

```diff
@@ -0,0 +1,372 @@
+            }
+        }
+
+        [Fact]
+        public void Test_Constructor_AzureOpenAI_InvalidAzureEndpoint()
+        {
+            // Arrange
+            var options = new AzureOpenAIModelOptions("test-key", "test-deployment", "https://test.openai.azure.com/");
+            options.AzureEndpoint = "test-endpoint";
+
+            // Act
+            Exception exception = Assert.Throws<ArgumentException>(() => new OpenAIModel(options));
+
+            // Assert
+            Assert.Equal("Model created with an invalid endpoint of `test-endpoint`. The endpoint must be a valid HTTPS url.", exception.Message);
+        }
+
+        [Fact]
+        public void Test_Constructor_AzureOpenAI_InvalidAzureApiVersion()
+        {
+            // Arrange
+            var options = new AzureOpenAIModelOptions("test-key", "test-deployment", "https://test.openai.azure.com/");
+            var versions = new List<string>
+            {
+                "2022-12-01", "2023-05-15", "2023-06-01-preview", "2023-07-01-preview", "2023-08-01-preview", "2023-09-01-preview"
+            };
+
+            // Act
+            foreach (var version in versions)
+            {
+                options.AzureApiVersion = version;
+                try
+                {
+                    new OpenAIModel(options);
+                }
+                catch (Exception ex)
+                {
+                    Assert.Fail($"Expected no exception, but got: {ex.Message}");
+                }
+            }
+            options.AzureApiVersion = "2023-12-01-preview";
+            Exception exception = Assert.Throws<ArgumentException>(() => new OpenAIModel(options));
+
+            // Assert
+            Assert.Equal("Model created with an unsupported API version of `2023-12-01-preview`.", exception.Message);
+        }
+
+        [Fact]
+        public async void Test_CompletePromptAsync_AzureOpenAI_Text_PromptTooLong()
+        {
+            // Arrange
+            var turnContextMock = new Mock<ITurnContext>();
+            var turnStateMock = new Mock<TurnState>();
+            var renderedPrompt = new RenderedPromptSection<string>(string.Empty, length: 65536, tooLong: true);
+            var promptMock = new Mock<Prompt>(new List<PromptSection>(), -1, true, "\n\n");
+            promptMock.Setup((prompt) => prompt.RenderAsTextAsync(
+                It.IsAny<ITurnContext>(), It.IsAny<IMemory>(), It.IsAny<IPromptFunctions<List<string>>>(),
+                It.IsAny<ITokenizer>(), It.IsAny<int>(), It.IsAny<CancellationToken>())).ReturnsAsync(renderedPrompt);
+            var promptTemplate = new PromptTemplate("test-prompt", promptMock.Object);
+            var options = new AzureOpenAIModelOptions("test-key", "test-deployment", "https://test.openai.azure.com/")
+            {
+                CompletionType = CompletionConfiguration.CompletionType.Text
+            };
+            var openAIModel = new OpenAIModel(options);
```

Comment:
Using `TestLoggerFactory` and setting `LogRequests` to true can increase a few coverage (log part)

Reply:
Good idea, added.